operators/src/processing/new_expression/codegen.rs

```diff
@@ -0,0 +1,353 @@
+                else_branch: default_branch,
+            } => {
+                let mut new_tokens = TokenStream::new();
+                for (i, branch) in condition_branches.iter().enumerate() {
+                    let condition = &branch.condition;
+                    let body = &branch.body;
+
+                    new_tokens.extend(if i == 0 {
+                        // first
+                        quote! {
+                            if #condition {
+                                #body
+                            }
+                        }
+                    } else {
+                        // middle
+                        quote! {
+                            else if #condition {
+                                #body
+                            }
+                        }
+                    });
+                }
+
+                new_tokens.extend(quote! {
+                    else {
+                        #default_branch
+                    }
+                });
+
+                new_tokens
+            }
+            Self::AssignmentsAndExpression {
+                assignments,
+                expression,
+            } => {
+                quote! {
+                    #(#assignments)*
+                    #expression
+                }
+            }
+        };
+
+        tokens.extend(new_tokens);
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+#[repr(transparent)]
+pub struct Identifier(String);
+
+impl ToTokens for Identifier {
+    fn to_tokens(&self, tokens: &mut TokenStream) {
+        let identifier = format_ident!("{}", self.0);
+        tokens.extend(quote! { #identifier });
+    }
+}
+
+impl From<String> for Identifier {
+    fn from(s: String) -> Self {
+        Self(s)
+    }
+}
+
+impl From<&str> for Identifier {
+    fn from(s: &str) -> Self {
+        Self(s.to_string())
+    }
+}
+
+impl From<&String> for Identifier {
+    fn from(s: &String) -> Self {
+        Self(s.to_string())
+    }
+}
+
+impl AsRef<str> for Identifier {
+    fn as_ref(&self) -> &str {
+        &self.0
+    }
+}
+
+impl std::fmt::Display for Identifier {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> Result<(), std::fmt::Error> {
+        std::fmt::Display::fmt(&self.0, f)
+    }
+}
+
+#[derive(Debug, Clone)]
+pub enum AstOperator {
+    Add,
+    Subtract,
+    Multiply,
+    Divide,
+}
+
+impl ToTokens for AstOperator {
+    fn to_tokens(&self, tokens: &mut TokenStream) {
+        let new_tokens = match self {
+            Self::Add => quote! { + },
+            Self::Subtract => quote! { - },
+            Self::Multiply => quote! { * },
+            Self::Divide => quote! { / },
+        };
+
+        tokens.extend(new_tokens);
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct Branch {
+    pub condition: BooleanExpression,
+    pub body: AstNode,
+}
+
+#[derive(Debug, Clone)]
+pub enum BooleanExpression {
+    Variable(Identifier),
+    Constant(bool),
+    Comparison {
```

Comment:
Isn't `BooleanExpression` your predicate? It can contain comparisons, a constant (bool), nested comparisons, …
just currently not yet any function call.